### PR TITLE
Migrate image works from Plum

### PR DESCRIPTION
**Pull request by eliotjordan**
Work towards migrating Image Works from Plum.

ToDo:
- [x] Bounding box widget
- [x] Multiple FileSet types - One for scanned map and one for external metadata
- [x] Metadata extraction

### CODE REVIEW

File: app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb [hunk 1]
```diff
@@ -13,13 +13,25 @@ def run
...
 
     def mint_ark?
       return false unless change_set.try(:new_state) == 'complete'
       change_set.try(:state_changed?) || change_set.apply_remote_metadata?
     end
+
+    private
+
+      def apply(attributes)
+        if change_set.respond_to?(:apply_remote_metadata_directly?) && change_set.apply_remote_metadata_directly?
```
Comment (tpendragon) [inline app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb]: Can we just make all the change_sets respond to this?

File: app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb [hunk 2]
```diff
@@ -13,13 +13,25 @@ def run
...
+          attributes.each do |key, value|
+            change_set.model.send("#{key}=".to_sym, value) if change_set.respond_to?(key)
```
Comment (tpendragon) [inline app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb]: Can you just do

```
change_set.validate(attributes)
change_set.sync
```
?

File: app/characterization_services/plum_characterization_service.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  def characterize(save: true)
+    TikaFileCharacterizationService.new(file_node: file_node, persister: persister).characterize
+  end
+
+  def valid?
+    !parent.is_a?(ScannedMap)
+  end
+
+  def parent
+    decorator = FileSetDecorator.new(file_node)
```
Comment (tpendragon) [inline app/characterization_services/plum_characterization_service.rb]: Does `file_node.decorate.parent` work?

File: app/models/controlled_vocabulary.rb [hunk 4]
```diff
@@ -98,6 +102,40 @@ def all(_scope = nil)
     end
   end
 
+  class GeoTerm < Valkyrie::Resource
```
Comment (tpendragon) [inline app/models/controlled_vocabulary.rb]: Why a special GeoTerm? Concerned about the returned thing from a ControlledVocabulary having different methods.

**Comment by tpendragon** [comment]
Also needs link on home page/dropdown menu to create a ScannedMap.

**Comment by tpendragon** [comment]
Missing structure editor link on show page of a ScannedMap.

File: app/models/controlled_vocabulary.rb (hunk 4 above)
Comment (eliotjordan) [reply app/models/controlled_vocabulary.rb]: I think that'll work. I'll just redefine `code` as `definition`.

File: app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb (hunk 2 above)
Comment (eliotjordan) [reply app/change_set_persisters/plum_change_set_persister/apply_remote_metadata.rb]: I see. Reform API. Yes.

**Comment by eliotjordan** [comment]
@tpendragon Addressed your changes and added scanned maps to the homepage.

**Pull request merged**